pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go

```diff
@@ -82,10 +83,20 @@ func (c *connectivityCheckController) Sync(ctx context.Context, syncContext fact
 
 	pnccClient := c.operatorcontrolplaneClient.ControlplaneV1alpha1().PodNetworkConnectivityChecks(operatorclient.TargetNamespace)
 	for _, check := range checks {
-		_, err := pnccClient.Get(ctx, check.Name, metav1.GetOptions{})
+		existing, err := pnccClient.Get(ctx, check.Name, metav1.GetOptions{})
 		if err == nil {
-			// already exists, skip
-			continue
+			if equality.Semantic.DeepEqual(existing.Spec, check.Spec) {
+				// already exists, no changes, skip
+				continue
+			}
+			updated := existing.DeepCopy()
+			updated.Spec = *check.Spec.DeepCopy()
+			_, err := pnccClient.Update(ctx, updated, metav1.UpdateOptions{})
+			if err != nil {
+				syncContext.Recorder().Warningf("EndpointDetectionFailure", "%s: %v", resourcehelper.FormatResourceForCLIWithNamespace(check), err)
+				continue
+			}
+			syncContext.Recorder().Eventf("EndpointCheckCreated", "Updated %s because it changed.", resourcehelper.FormatResourceForCLIWithNamespace(check))
```

Comment:
these events are going to the wrong namespace.  We want them in the `TargetNamespace`

Reply:
> these events are going to the wrong namespace. We want them in the `TargetNamespace`

will take a bug to fix it instead of fixing it here

Reply:
> these events are going to the wrong namespace. We want them in the `TargetNamespace`

@deads2k  Are you sure? These are events from the operator, managing the `Spec` of the **PodNetworkConnectivityChecks** (similar to events creating other resources in target namespace).  Compared this to the events coming from the **check-endpoints** tool that runs in the target namespace and updates the `Status` of the **PodNetworkConnectivityChecks**.

Reply:
it wasn't where I looked for them, but I guess I can live with it now that I know.